src/tree/map.rs

```diff
@@ -173,18 +173,18 @@ impl<K, V, C> Default for TreeMap<K, V, C> where C: Compare<K> + Default {
     fn default() -> TreeMap<K, V, C> { TreeMap::with_comparator(Default::default()) }
 }
 
-impl<K, V, C, Q: ?Sized> ops::Index<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
+impl<K, V, C, Q: Sized> ops::Index<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
```

Comment:
`TreeMap` has actually moved to https://github.com/contain-rs/bst, so rather than duplicate these changes here, could you just remove `src/tree` in a separate commit?